tck/src/main/java/org/eclipse/microprofile/concurrency/tck/MPConfigTest.java

```diff
@@ -405,6 +411,81 @@ public void overrideContextPropagationForThreadContextParameterWithConfig() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+    /**
+     * Verify MicroProfile Config overrides the propagated attribute to enabled
+     * an implied clear of all remaining.
+     * @throws TimeoutException 
+     * @throws ExecutionException 
+     * @throws InterruptedException 
+     */
+    @Test
+    public void overrideContextPropagationForThreadContextWithImpliedCleared() throws InterruptedException, ExecutionException, TimeoutException {
+        ExecutorService unmanagedThreads = null;
+        
+        // Expected config is propagated=Buffer; cleared=Remaining; unchanged=Label
+        Assert.assertNotNull(clearAllRemainingThreadContext,
+                "Unable to inject ThreadContext qualified by NamedInstance. Cannot run test.");
+        
+        int originalPriority = Thread.currentThread().getPriority();     
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 3 ? 2 : 3;
+            Thread.currentThread().setPriority(newPriority);
+            Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-A"));
+            Label.set("clearUnspecifiedContexts-test-label-A");
+
+            Callable<Integer> callable = clearAllRemainingThreadContext.contextualCallable(() -> {
+                try {
+                    Assert.assertEquals(Buffer.get().toString(), "clearUnspecifiedContexts-test-buffer-A",
+                            "Context type was not propagated to contextual action.");
+
+                    Assert.assertEquals(Label.get(), "",
+                            "Context type was not left unchanged by contextual action.");
```

Comment:
Using an executor service to run on a thread with a cleared Label isn't a good combination for this test because it prevents you from fully testing that Label is unchanged per the configuration.  There should be no need to create an executor service here, because it also makes the test unnecessarily complicated as well as interfering here with verifying results.  Instead, once the contextualCallable method finishes, you can update the context on the main test thread to whatever you want before running it.  For example, to clearUnspecifiedContexts-test-buffer-C and clearUnspecifiedContexts-test-label-C.  Then run the contextualCallable inline, which will let you verify that the label was left unchanged on the thread.